### PR TITLE
GM switching profile & GW profile & service data sources

### PR DESCRIPTION
**Pull request by asarfaty**

### CODE REVIEW

File: nsxt/data_source_nsxt_policy_qos_profile_test.go [hunk 1]
```diff
@@ -72,20 +92,38 @@ func testAccDataSourceNsxtPolicyQosProfileDeleteByName(name string) error {
...
+		if err != nil {
+			return handleListError("QosProfile", err)
+		}
+		for _, objInList := range objList.Results {
+			if *objInList.DisplayName == name {
+				err := client.Delete(*objInList.Id)
+				if err != nil {
+					return handleDeleteError("QosProfile", *objInList.Id, err)
+				}
+				return fmt.Errorf("DEBUG ADIT this is global manager")
```
Comment (annakhm) [inline nsxt/data_source_nsxt_policy_qos_profile_test.go:109]: letfover

File: nsxt/data_source_nsxt_policy_qos_profile_test.go [hunk 2]
```diff
@@ -72,20 +92,38 @@ func testAccDataSourceNsxtPolicyQosProfileDeleteByName(name string) error {
...
-	if err != nil {
-		return handleListError("QosProfile", err)
-	}
-	for _, objInList := range objList.Results {
-		if *objInList.DisplayName == name {
-			err := client.Delete(*objInList.Id)
-			if err != nil {
-				return handleDeleteError("QosProfile", *objInList.Id, err)
+	if testAccIsGlobalManager() {
+		client := gm_infra.NewDefaultQosProfilesClient(connector)
```
Comment (annakhm) [inline nsxt/data_source_nsxt_policy_qos_profile_test.go:98]: do you think we should replace this with search function as well?

File: nsxt/data_source_nsxt_policy_qos_profile_test.go (hunk 1 above)
Comment (asarfaty) [reply nsxt/data_source_nsxt_policy_qos_profile_test.go:109]: Done

File: nsxt/data_source_nsxt_policy_qos_profile_test.go (hunk 2 above)
Comment (asarfaty) [reply nsxt/data_source_nsxt_policy_qos_profile_test.go:98]: Not sure what you mean. There is not default QoS profile that we can use here, so forced to create it.

Comment (asarfaty) [reply nsxt/data_source_nsxt_policy_qos_profile_test.go:98]: Done

**Pull request merged**